File: sketch/src/main/java/com/github/panpf/sketch/SketchException.java
package com.github.panpf.sketch;

public abstract class SketchException extends Exception {
    public SketchException() {
    }

    public SketchException(String message) {
        super(message);
    }

    public SketchException(String message, Throwable cause) {
        super(message, cause);
    }

    public SketchException(Throwable cause) {
        super(cause);
    }
}

File: sketch-core/src/main/java/com/github/panpf/sketch/request/SketchException.kt
package com.github.panpf.sketch.request

abstract class SketchException(
    message: String,
    cause: Throwable? = null
) : Exception(message, cause)

class DecodeException(
    message: String,
    cause: Throwable? = null
) : SketchException(message, cause)